server/hotrod/src/main/java/org/infinispan/server/hotrod/TransactionRequestProcessor.java

```diff
@@ -195,39 +190,50 @@ private void validateConfiguration(AdvancedCache<byte[], byte[]> cache) {
     * If the transaction isn't prepared and the originator left the cluster, the previous transaction is rolled-back and
     * a new one is started.
     */
-   private boolean checkExistingTxForPrepare(HotRodHeader header, PrepareTransactionDecodeContext context) {
-      TxState txState = context.getTxState();
+   private boolean checkExistingTxForPrepare(HotRodHeader header, PrepareCoordinator txCoordinator) {
+      TxState txState = txCoordinator.getTxState();
       if (txState == null) {
          return false;
       }
-      switch (txState.status()) {
-         case Status.STATUS_ACTIVE:
-            break;
-         case Status.STATUS_PREPARED:
+      if (txCoordinator.isAlive(txState.getOriginator())) {
+         //transaction started on another node but the node is still in the topology. 2 possible scenarios:
+         // #1, the topology isn't updated
+         // #2, the client timed-out waiting for the reply
+         //in any case, we send a ignore reply and the client is free to retry (or rollback)
+         writeNotExecuted(header);
+         return true;
+      }
+      //originator is dead...
+
+      //First phase state machine
+      //success ACTIVE -> PREPARING -> PREPARED
+      //failed ACTIVE -> MARK_ROLLBACK -> ROLLED_BACK or ACTIVE -> PREPARING -> ROLLED_BACK
+      //1PC success ACTIVE -> PREPARING -> MARK_COMMIT -> COMMITTED
```

Comment:
Need some more comments on these, I'm sure what MARK_COMMIT is for, and I'm starting to doubt that I know what PREPARING means as well :)

Reply:
PREPARING means the transaction has invoked tx.runPrepare()
MARK_COMMIT / ROLLBACK means that we know the client's TM decision and we are performing the commit / rollback.

Reply:
I'm going to add javadoc to the Status enum